### PR TITLE
[Merged by Bors] - Move rust code to top level

### PR DESCRIPTION
**Pull request by acrrd**
Move rust code from ` discovery_engine_core` to the top level of the repository.
Remove RUST_WORKSPACE env variable.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,5 @@
 .env.local
 
 headlines_download/
+target/
+.env.db.dev
```
Comment (acrrd) [inline .gitignore:11]: This are from `discovery_engine_core/.gitignore`.

File: justfile [hunk 2]
```diff
@@ -92,12 +85,11 @@ test: rust-test
 
 # Cleans up rusts build cache
 rust-clean:
-    cd "$RUST_WORKSPACE"; \
     cargo clean
 
 # Removes all asset data
 remove-assets:
-    find $RUST_WORKSPACE/assets/*_v* -exec rm -fr '{}' \;
```
[the remark this answers is not in the document]
Comment (janpetschexain) [reply justfile:100]: :question: won't this also delete the `.gitignore` file in that dir?

**Review by janpetschexain — COMMENTED** [review]
lgtm, but there are some leftovers

:wrench: in the `build-ci-image.yml` workflow the path to the toolchain file needs to be updated in l.36

:wrench: in the `pull-request.yml` workflow the path for the filter needs to be adjusted in l.33

:wrench:  in the `onnx-tool/Readme.md` the path to the assets needs to be adjusted in l.25

File: justfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (acrrd) [reply justfile:100]: No it only matches the non hidden files.

**Comment by acrrd** [comment]
Test run of the "build" CIs
https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/3480642914
https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/3480654188
https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/3480660743

**Comment by acrrd** [comment]
https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/3480777122
https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/3480781726

**Comment by acrrd** [comment]
https://github.com/xaynetwork/xayn_discovery_engine/actions/runs/3480817650

**Review by Robert-Steiner — COMMENTED** [review]
the changes in `web-services.yml` and `ingestion-service` lgtm

**Review by janpetschexain — APPROVED** [review]
👍

**Comment by acrrd** [comment]
bors merge

**Pull request closed without merging**